pkg/oob/oob.go

```diff
@@ -36,77 +36,89 @@ type BMCResetter interface {
 // SetPower interface function for power actions
 func SetPower(ctx context.Context, action string, m []PowerSetter) (result string, err error) {
 	for _, elem := range m {
-		result, setErr := elem.PowerSet(ctx, action)
-		if setErr != nil {
-			err = multierror.Append(err, setErr)
-			continue
+		if elem != nil {
+			result, setErr := elem.PowerSet(ctx, action)
+			if setErr != nil {
+				err = multierror.Append(err, setErr)
+				continue
+			}
+			return result, nil
 		}
-		return result, nil
 	}
 	return result, multierror.Append(err, errors.New("power state failed"))
```

Comment:
Would it be better to provide a more detailed error when elem is nil, such as `power request not executed` or something?

Reply:
good idea, will update.